### PR TITLE
feat: improve performance for multithreaded uploading

### PR DESCRIPTION
**Pull request by arvgord**
- Improved performance for multithreaded uploading

For better performance user should upload files to a DICOM Store folder, not a Series folder.
Also for better performance uploaded files should be with an extension(.dcm - etc).

### CODE REVIEW

**Comment by dzelemba** [comment]
This is a very large PR... can it be split up into smaller ones? E.g. split out supporting navigation vs. improved multithreading performance. Or maybe do the refactor first in a separate PR and then follow up with these.

**Comment by arvgord** [comment]
Ok, I'll split.

**Comment by arvgord** [comment]
> This is a very large PR... can it be split up into smaller ones? E.g. split out supporting navigation vs. improved multithreading performance. Or maybe do the refactor first in a separate PR and then follow up with these.

Done. 
I renamed DicomFuseHelperTest to DicomFuseHelperIT because this test is more of the integration test than the unit test. In addition, I refactored the DicomFuseHelperIT to get a more realistic test.

File: src/test/java/com/google/dicomwebfuse/fuse/DicomFuseHelperIT.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+import com.google.dicomwebfuse.exception.DicomFuseException;
+import com.google.dicomwebfuse.fuse.cacher.DicomPathCacher;
+import com.google.dicomwebfuse.parser.Arguments;
+import java.io.IOException;
+import jnr.ffi.Platform;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class DicomFuseHelperIT {
```
Comment (dzelemba) [inline src/test/java/com/google/dicomwebfuse/fuse/DicomFuseHelperIT.java]: Let's keep this named DicomFuseHelperTest... anything that doesn't require any external resources I consider a unit test.

Comment (arvgord) [reply src/test/java/com/google/dicomwebfuse/fuse/DicomFuseHelperIT.java]: Done

**Comment by arvgord** [comment]
> Can you explain what changes are being made and why in the commit description?

Done.

> A summary of how much this improves performance would also be useful.

I've presented short info in the commit. I'll provide detailed performance info on the Wiki page.

**Comment by arvgord** [comment]
@dzelemba Could you review, please?

**Pull request merged**